sim/mage/combustion.go

```diff
@@ -36,6 +36,9 @@ func (mage *Mage) registerCombustionSpell() {
 				spell.DealDamage(sim, result)
 				spell.RelatedDotSpell.Cast(sim, target)
 			}
+			if mage.t13ProcAura != nil {
+				spell.CD.Reduce(time.Second * time.Duration(5*mage.t13ProcAura.GetStacks()))
+			}
```

Comment:
This should go in the `OnStacksChange` callback for `mage.t13ProcAura` rather than in the `ApplyEffects` for the affected spell. With your current implementation, if more stacks are gained after Combustion is cast, the CD won't be reduced further when it should.

Reply:
From what I've read it calculates it on casting the sprll, not after/during the uptime. So any stacks after casting this do not count anymore.